x-pack/plugins/snapshot_restore/public/app/components/repository_type_name.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  AzureRepositoryType,
+  FSRepositoryType,
+  GCSRepositoryType,
+  HDFSRepositoryType,
+  ReadonlyRepositoryType,
+  RepositoryType,
+  S3RepositoryType,
+  SourceRepositoryType,
+} from '../../../common/repository_types';
+import { AppStateInterface, useAppState } from '../services/app_context';
+
+interface Props {
+  type: RepositoryType;
+  delegateType?: RepositoryType;
+}
+
+export const RepositoryTypeName = ({ type, delegateType }: Props) => {
+  const [
+    {
+      core: {
+        i18n: { FormattedMessage },
+      },
+    },
+  ] = useAppState() as [AppStateInterface];
+
+  const getTypeName = (repositoryType: RepositoryType): JSX.Element => {
+    switch (repositoryType) {
```

Comment:
Another minor nit, you could use a map instead of a switch to reduce visual noise a bit:

```js
    const map = {
      [FSRepositoryType]: (
        <FormattedMessage
          id="xpack.snapshotRestore.repositoryType.fileSystemTypeName"
          defaultMessage="Shared File System"
        />
      ),

      [ReadonlyRepositoryType]: (
        <FormattedMessage
          id="xpack.snapshotRestore.repositoryType.readonlyTypeName"
          defaultMessage="Read-only URL"
        />
      ),

      [S3RepositoryType]: (
        <FormattedMessage
          id="xpack.snapshotRestore.repositoryType.s3TypeName"
          defaultMessage="AWS S3"
        />
      ),

      /* ... */      
    };
```